subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/delegates/ProjectDelegate.kt

```diff
@@ -103,6 +103,8 @@ abstract class ProjectDelegate : Project {
     override fun getLogger(): Logger =
         delegate.logger
 
+    @Suppress("DEPRECATION")
+    @Deprecated("Use layout.buildDirectory instead", ReplaceWith("layout.buildDirectory.get()"))
     override fun getBuildDir(): File =
         delegate.buildDir
```

Comment:
❔ Can't the deprecation suppression be reduced to the call to the delegate as in `setBuildDir` below?

```suggestion
    @Deprecated("Use layout.buildDirectory instead", ReplaceWith("layout.buildDirectory.get()"))
    override fun getBuildDir(): File =
        @Suppress("DEPRECATION")
        delegate.buildDir
```